lib/ruby_llm/provider.rb

```diff
@@ -60,8 +60,10 @@ def complete(messages, tools:, temperature:, model:, params: {}, headers: {}, sc
     end
 
     def list_models
-      response = @connection.get models_url
-      parse_list_models_response response, slug, capabilities
+      Array(models_url).flat_map do |url|
+        response = @connection.get(url)
+        parse_list_models_response(response, slug, capabilities)
+      end
```

Comment:
Let's not change the API there. Since only Grok needs multiple URLs just override `list_models` in the XAI provider.